pkgs/development/compilers/flutter/engine/package.nix

```diff
@@ -312,7 +312,7 @@ stdenv.mkDerivation (finalAttrs: {
   installPhase = ''
     runHook preInstall
 
-    rm -rf $out/out/$outName/{obj,gen,exe.unstripped,lib.unstripped,zip_archives}
+    rm -rf $out/out/$outName/{obj,exe.unstripped,lib.unstripped,zip_archives}
```

Comment:
i don't think so? for one, you *can* mix and match "engine built as `host_release`" and "app built with `flutter build --debug`", which produces an error message complaining about the nonexistence of `/nix/store/ac3bxkinmi0210zx8cxqxxbq4hq2p8p3-flutter-engine-36335019a8eab588c3c2ea783c618d90505be233/out/host_release/gen/flutter/lib/snapshot/vm_isolate_snapshot.bin`.

in general though, it's not clear that this is only used in debug builds? yes, in the specific issue i had, this only affects debug builds (of the app! not of the engine!), so it could make sense to only have it for that build configuration, but as you said, there are no packaging instructions, and *by the looks of it*, this directory is in general important and supposed to be kept. i *think* it's more correct to keep it always, but i don't actually know a use case where any part of it other than `gen/flutter/lib/snapshot/*` matters.

Reply:
> for one, you _can_ mix and match "engine built as `host_release`" and "app built with `flutter build --debug`", which produces an error message complaining about the nonexistence of `/nix/store/ac3bxkinmi0210zx8cxqxxbq4hq2p8p3-flutter-engine-36335019a8eab588c3c2ea783c618d90505be233/out/host_release/gen/flutter/lib/snapshot/vm_isolate_snapshot.bin`.

Hmm, I thought you can only use a debug engine to run debug snapshots and release engines can only run release snapshots?

Reply:
I'd like to share a project I found called `cargo flutter`, which integrates with Flutter and Cargo to provide a way to build embedded apps all in one.

It has absolutely no inherent knowledge of the engine build profiles. In fact, it seems to primarily want these for cross compilation? because it's working with two of them at any given time. But the build profile is a completely different parameter. There's also a not-exactly-secret third option called `--profile` which doesn't clearly align with a flavor of the engine.

It also makes use of [a `frontend_server` present in this `/gen/` directory](https://github.com/flutter-rs/cargo-flutter/blob/017b71f70339da5c227464c1c2cb944f61376052/src/flutter.rs#L100-L103), which is required to compile a Dart/Flutter project ahead of time (required for release builds).

So, while it can mix and match with any engine as mentioned, it is the case that even if we only consider {release engine, release app} and {debug engine, debug app}, then:

- the issue I reported requires a `/gen/` artifact in debug mode, *and*
- building an AOT executable (as seen in e.g. `cargo-flutter`) requires a `/gen/` artifact in release mode

so no, I don't think it should be pruned for either of our engine build profiles (is there a standard name for this? it's confusing to clearly disambiguate from the app build profile, and also from like, a `--profile` build)